### PR TITLE
add pfSense tftp package for tftp-hpa daemon

### PR DESCRIPTION
**Pull request by seidler2547**
Rather than porting the old tftp package, I created a new package for the much more versatile and fast tftp-hpa daemon. This is my first attempt at packaging, so please advise about any problems with the code. The package did compile, install and work correctly on a fresh pfSense 2.3.2 install.

### CODE REVIEW

**Comment by seidler2547** [comment]
Further testing also showed the package to be working on a fresh i386 install and a i386 install which was upgraded from 2.2.6 (originally 2.1.x) to 2.3.2 (AMD T40E PC-Engines APU board).

File: ftp/pfSense-pkg-tftpd/Makefile [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# $FreeBSD$
+
+PORTNAME=	pfSense-pkg-tftpd
+PORTVERSION=	0.1.2
+PORTREVISION=	1
```
Comment (rbgarga) [inline ftp/pfSense-pkg-tftpd/Makefile]: PORTREVISION line should be removed from new ports

File: ftp/pfSense-pkg-tftpd/files/etc/inc/priv/tftpd.priv.inc [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+	INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+	AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+	AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+	OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+	SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+	INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+	CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+	ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+	POSSIBILITY OF SUCH DAMAGE.
```
Comment (rbgarga) [inline ftp/pfSense-pkg-tftpd/files/etc/inc/priv/tftpd.priv.inc]: We moved to APACHE20 license, please replace it accodingly, you can see an example at
https://github.com/pfsense/FreeBSD-ports/blob/devel/security/pfSense-pkg-sudo/files/etc/inc/priv/sudo.priv.inc#L2

File: ftp/pfSense-pkg-tftpd/files/usr/local/pkg/tftpd.inc [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+	INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+	AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+	AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+	OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+	SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+	INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+	CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+	ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+	POSSIBILITY OF SUCH DAMAGE.
```
Comment (rbgarga) [inline ftp/pfSense-pkg-tftpd/files/usr/local/pkg/tftpd.inc]: Replace license text by current one

File: ftp/pfSense-pkg-tftpd/files/usr/local/pkg/tftpd.inc [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+function deinstall_package_tftpd() {
+	rmdir("/tftpboot");
+}
+
+function sync_package_tftpd() {
+	global $config;
+
+	conf_mount_rw();
```
Comment (rbgarga) [inline ftp/pfSense-pkg-tftpd/files/usr/local/pkg/tftpd.inc]: nanobsd support was removed, you can remove all calls to conf_mount_rw and conf_mount_ro

File: ftp/pfSense-pkg-tftpd/files/usr/local/pkg/tftpd.inc [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+		sleep(3);
+	}
+	/* Only (re)start the service when it is enabled */
+	if ($tftpd_conf['tftpd_enable'] == "on") {
+		start_service("tftpd");
+		sleep(3);
+	}
+	filter_configure();
+
+	conf_mount_ro();
```
Comment (rbgarga) [inline ftp/pfSense-pkg-tftpd/files/usr/local/pkg/tftpd.inc]: Remove call to conf_mount_ro

File: ftp/pfSense-pkg-tftpd/files/usr/local/pkg/tftpd.xml [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+	AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+	AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+	OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+	SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+	INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+	CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+	ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+	POSSIBILITY OF SUCH DAMAGE.
+*/
+/* ====================================================================================== */
```
Comment (rbgarga) [inline ftp/pfSense-pkg-tftpd/files/usr/local/pkg/tftpd.xml]: Replace license text

File: ftp/pfSense-pkg-tftpd/Makefile [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+CATEGORIES=	ftp
+MASTER_SITES=	# empty
+DISTFILES=	# empty
+EXTRACT_ONLY=	# empty
+
+MAINTAINER=	pfsense@stefanseidel.info
+COMMENT=	pfSense package for tftp server
+
+LICENSE=	GPLv3
```
Comment (rbgarga) [inline ftp/pfSense-pkg-tftpd/Makefile]: This code is APACHE20

**Comment by seidler2547** [comment]
Ok, I think I have made all the requested changes.

**Pull request merged**